staging/src/k8s.io/client-go/util/flowcontrol/backoff_test.go

```diff
@@ -250,3 +250,34 @@ func TestBackoffWithJitter(t *testing.T) {
 
 	t.Logf("exponentially backed off jittered delays: %v", delays)
 }
+
+func TestAlternateHasExpiredFunc(t *testing.T) {
+	id := "_alternateHasExpiredFunc"
+	tc := testingclock.NewFakeClock(time.Now())
+	step := time.Second
+	maxDuration := 50 * step
+
+	b := NewFakeBackOff(step, maxDuration, tc)
+
+	// try several short (<maxduration) expirations that we can test with
+	cases := []time.Duration{1, 2, 4, 8}
+
+	for _, c := range cases {
+		b.HasExpiredFunc = func(eventTime time.Time, lastUpdate time.Time, maxDuration time.Duration) bool {
+			return eventTime.Sub(lastUpdate) < (c * step)
+		}
+
+		// increment time past expected hasexpired value and force backoff to check
+		tc.Step(c * step)
+		b.Next(id, tc.Now())
+
+		w := b.Get(id)
+		if w != step {
+			t.Errorf("hasexpired value: '%d': expected %s (backoff to be reset to initial), got %s", c, step, w)
+		}
```

Comment:
Verify that it hasn't expired before stepping the time.

Reply:
Fixed in https://github.com/kubernetes/kubernetes/commit/815001e983e700b053f236bcea49b794f20a65dc and found the most rookie bug in doing so :)